File: clients/java/examples/tables-list-types.java
import com.datastax.astra.client.DataAPIClient;
import com.datastax.astra.client.databases.Database;
import com.datastax.astra.client.tables.definition.types.TableUserDefinedTypeDescriptor;
import java.util.List;

public class Example {

  public static void main(String[] args) {
    // Get a database
    Database database = new DataAPIClient("**APPLICATION_TOKEN**").getDatabase("**API_ENDPOINT**");

    // List type metadata
    List<TableUserDefinedTypeDescriptor> result = database.listTypes();

    System.out.println(result);
  }
}

File: clients/typescript/examples/tables-list-types.ts
import { DataAPIClient } from "@datastax/astra-db-ts";

// Get a database
const client = new DataAPIClient();
const database = client.db("**API_ENDPOINT**", {
  token: "**APPLICATION_TOKEN**",
});

// List type metadata
(async function () {
  const result = await database.listTypes({ nameOnly: false });

  console.log(result);
})();
